app/src/org/commcare/recovery/measures/ScanCczTask.java

```diff
@@ -94,7 +97,19 @@ private void locateAppCcz(File[] files, int depth) {
         String lastKnownCczLocation = HiddenPreferences.getLastKnownCczLocation();
         ArrayList<File> filePathsToScan = new ArrayList<>(3);
         if (!StringUtils.isEmpty(lastKnownCczLocation)) {
-            filePathsToScan.add(new File(lastKnownCczLocation));
+            try {
+                if (lastKnownCczLocation.startsWith("content://")) {
+                    InputStream inputStream = CommCareApplication.instance().getContentResolver().openInputStream(Uri.parse(lastKnownCczLocation));
+                    File destination = new File(CczUtils.getCczTargetPath());
+                    FileUtil.copyFile(inputStream, destination);
+                    filePathsToScan.add(destination);
+                } else {
+                    filePathsToScan.add(new File(lastKnownCczLocation));
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+                Logger.log(LogTypes.TYPE_ERROR_STORAGE, "Last known ccz location failed with : " + e.getMessage());
+            }
         }
         filePathsToScan.add(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS));
```

Comment:
I think these paths will also throw the security errors right ?

Reply:
Yeah, I think so.